yarn-plugin/sources/commands/spdx.ts

```diff
@@ -136,11 +113,19 @@ export class SpdxCommand extends BaseCommand {
         const dependencyPkg = project.storedPackages.get(identHash);
         if (!dependencyPkg) continue;
 
-        spdxDocument.addRelationship(
-          currentPkgSpdxId,
-          getSpdxId(dependencyPkg),
-          spdxDependsOn,
-        );
+        if (devDependencies.has(pkg.identHash)) {
```

Comment:
Both of the points raised are absolutely correct. Therefore, the original approach was not valid.
It's rewritten now, so that those points are addressed.